packages/core/src/Badge/Badge.tsx

```diff
@@ -57,11 +57,73 @@ const sizes = {
   `,
 }
 
+const badgeStyle = (props) => {
+  const badgeStyles = {
+    primary: {
+      backgroundColor: props.theme.palette.primary.base,
+      color: props.theme.palette.text.lightest,
+    },
+    primaryLight: {
+      backgroundColor: props.theme.palette.primary.light,
+      color: props.theme.palette.primary.base,
+    },
+    primaryDark: {
+      backgroundColor: props.theme.palette.primary.dark,
+      color: props.theme.palette.text.lightest,
+    },
+    primaryDarkLight: {
+      backgroundColor: props.theme.palette.primary.light,
+      color: props.theme.palette.primary.dark,
+    },
+    secondary: {
+      backgroundColor: props.theme.palette.secondary.base,
+      color: props.theme.palette.text.lightest,
+    },
+    secondaryLight: {
+      backgroundColor: props.theme.palette.secondary.light,
+      color: props.theme.palette.secondary.base,
+    },
+    backgroundDarkest: {
+      backgroundColor: props.theme.palette.background.darkest,
+      color: props.theme.palette.text.lightest,
+    },
+    backgroundLighest: {
+      backgroundColor: props.theme.palette.background.lightest,
+      color: props.theme.palette.text.lightest,
+    },
+    neutral: {
+      backgroundColor: props.theme.palette.background.dark,
+      color: props.theme.palette.text.lightest,
+    },
+    neutralLight: {
+      backgroundColor: props.theme.palette.background.dark,
+      color: props.theme.palette.text.lightest,
+    },
+    warning: {
+      backgroundColor: props.theme.palette.warning.base,
+      color: props.theme.palette.text.lightest,
+    },
+    warningLight: {
+      backgroundColor: props.theme.palette.warning.light,
+      color: props.theme.palette.warning.base,
+    },
+    highlight: {
+      backgroundColor: props.theme.palette.highlight.light,
+      color: props.theme.palette.highlight.shade,
+    },
+    notify: {
+      backgroundColor: props.theme.palette.notify.light,
+      color: props.theme.palette.text.base,
+    },
+  }
+  return !props.bg && !props.color && (badgeStyles[props.badgeStyle] || {})
```

Comment:
I wanted to start a discussion based on this opinion of mine.  Here I'd like to handle the case of when a consumer assigns a value to `bg`, `color` and `badgeStyle`, by making `badgeStyle` a no-op.  Ideally, I think it'll be thoughtful to raise a warning in their browser like we do with [deprecatedColorValue](https://github.com/priceline/design-system/pull/1265/files#diff-deaba2548643a017885d124c11f77e7d967dfa8cfdac98c6a8c6710e27b9d1e7L83)

Reply:
I think that's a good idea, although most people tend to tune out DS warnings. I wonder if throwing an error would be too obtrusive.

Reply:
Agree with the order of precedence, where badgeStyle is top-level important. I think a warning might be enough, because it will be so visually noticeable